framework/infrastructure/gcp/cloud_run.py

```diff
@@ -0,0 +1,147 @@
+        self.project = project
+        self.region = region
+        service: discovery.Resource = self.api_manager.cloudrun("v2")
+        self.service = service
+        self._parent = f"projects/{self.project}/locations/{self.region}"
+        super().__init__(self.service, project)
+
+    @property
+    def api_name(self) -> str:
+        """Returns the API name for Cloud Run."""
+        return "run"
+
+    @property
+    def api_version(self) -> str:
+        """Returns the API version for Cloud Run."""
+        return "v2"
+
+    def create_service(
+        self, service: discovery.Resource, service_name: str, body: dict
+    ) -> GcpResource:
+        return self._create_resource(
+            collection=service.projects().locations().services(),
+            location=self.region,
+            serviceId=service_name,
+            body=body,
+        )
+
+    def get_service(
+        self, service: discovery.Resource, service_name: str
+    ) -> CloudRun:
+        result = self._get_resource(
+            collection=service.projects().locations().services(),
+            full_name=self.resource_full_name(
+                service_name, "services", self.region
+            ),
+        )
+        return CloudRun.from_response(
+            self.resource_full_name(service_name, "services", self.region),
+            result,
+        )
+
+    def get_service_uri(self, service_name: str) -> str:
+        response = self.get_service(self.service, service_name)
+        return response.url
+
+    def delete_service(self, service_name: str):
+        # pylint: disable=no-member
+        self._delete_resource(
+            self.service.projects().locations().services(),
+            full_name=self.resource_full_name(
+                service_name, "services", self.region
+            ),
+        )
+
+    def deploy_service(
+        self,
+        service_name: str,
+        image_name: str,
+        *,
+        test_port: int = DEFAULT_TEST_PORT,
+    ):
+        if not service_name:
+            raise ValueError("service_name cannot be empty or None")
+        if not image_name:
+            raise ValueError("image_name cannot be empty or None")
+
+        try:
+            service_body = {
+                "launch_stage": "alpha",
+                "template": {
+                    "containers": [
+                        {
+                            "image": image_name,
+                            "ports": [
+                                {"containerPort": test_port, "name": "h2c"}
+                            ],
+                        }
+                    ],
+                },
+            }
+
+            logger.info("Deploying Cloud Run service '%s'", service_name)
+            self.create_service(self.service, service_name, service_body)
+            return self.get_service_uri(service_name)
+
+        except Exception as e:  # noqa pylint: disable=broad-except
```

Comment:
Here and elsewhere: avoid broad exceptions. I'm not sure we even need to catch anything here, just let the error propagate as is